### PR TITLE
Fix/revision history

### PR DESCRIPTION
**Pull request by valentindey**
full support for integer versioning, enable summary propagation, some fixes and more tests

### CODE REVIEW

File: documents/architecture-decisions.md [hunk 1]
```diff
@@ -648,34 +648,33 @@ As it stands, we have three different logics, how version numbers are
...
-     (analog 2) )
+   - The "draft counter" is added immediately when creating the new "draft"
+     version (-1) and thereafter the draft index is incremented with each save
+     (analogous to 2.) )
 
 for versions > 1.0.0
-Here we will add the "draft counter" immediately when creating the new "draft"
-version (-1) and thereafter increment the draft index with each save (analog 2) )
+The "draft counter" is added immediately when creating the new "draft"
+version (-1) and thereafter the draft index is incremented with each save
```
Comment (cloeser) [inline documents/architecture-decisions.md:676]: "-1.0" ?

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java [hunk 2]
```diff
@@ -444,68 +472,84 @@ public AdvisoryWrapper addEntryForNewCreatedVersion(String summary, String legac
...
+        return lastHistoryNode.get("summary").asText();
     }
 
-    public void removeAllPrereleaseVersions() {
-
-        if (getVersioningStrategy().getVersioningType() == VersioningType.Semantic && isInitialPublicReleaseOrEarlier()) {
-            ArrayNode historyNode = getOrCreateHistoryNode();
-            historyNode.removeAll();
-        }
+    public AdvisoryWrapper setLastRevisionHistoryEntryNumber(String newNumber) {
```
Comment (cloeser) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java:502]: I think "setLastRevisionHistoryElementNumber" is more clear.

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java [hunk 3]
```diff
@@ -444,68 +472,84 @@ public AdvisoryWrapper addEntryForNewCreatedVersion(String summary, String legac
...
+        ObjectNode lastHistoryNode = getLastHistoryNodeByDate();
+        lastHistoryNode.put("number", newNumber);
+        return this;
     }
 
-    private ObjectNode getLatestEntryInHistoryAfterPrerelease(ArrayNode historyNode) {
+    private ObjectNode getLastHistoryNodeByDate() {
```
Comment (cloeser) [inline src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java:508]: Here we call it "Node", maybe we can use a consistent wording. Be it "Node", "Element" or something else.

Comment (valentindey) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java:508]: I updated it to consistently use "element".

File: src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java (hunk 2 above)
Comment (valentindey) [reply src/main/java/de/bsi/secvisogram/csaf_cms_backend/json/AdvisoryWrapper.java]: updated

File: documents/architecture-decisions.md (hunk 1 above)
Comment (valentindey) [reply documents/architecture-decisions.md]: updated

File: documents/architecture-decisions.md [hunk 4]
```diff
@@ -648,34 +648,33 @@ As it stands, we have three different logics, how version numbers are
...
+     version and major, minor, or patch version are incremented by 1 accordingly,
+     however a higher level change resets the lower level change back to 0 and
+     each level can be incremented by 1 at most
+     → 1.0.0 can become only: 2.0.0, 1.1.0 or 1.0.1
 
-   - Here we will add the "draft counter" immediately when creating the new "draft"
-     version (-1) and thereafter increment the draft index with each save
-     (analog 2) )
+   - The "draft counter" is added immediately when creating the new "draft"
+     version (-1) and thereafter the draft index is incremented with each save
```
[the remark this answers is not in the document]
Comment (valentindey) [reply documents/architecture-decisions.md]: updated

File: documents/architecture-decisions.md [hunk 5]
```diff
@@ -648,34 +648,33 @@ As it stands, we have three different logics, how version numbers are
...
-   - It is checks what type of change was made compared to thet las published
-     version and increments major, minor, or patch by 1 accordingly, however a
-     higher level change resets the lower level change back to 0 and each level
-     can be incremented by 1 at most
-     → 1.0.0 can become only: 2.0.0, 1.1.0, 1.0.1
+   - It is checked what type of change was made compared to the last published
+     version and major, minor, or patch version are incremented by 1 accordingly,
+     however a higher level change resets the lower level change back to 0 and
+     each level can be incremented by 1 at most
+     → 1.0.0 can become only: 2.0.0, 1.1.0 or 1.0.1
```
[the remark this answers is not in the document]
Comment (valentindey) [reply documents/architecture-decisions.md]: updated

File: documents/architecture-decisions.md [hunk 6]
```diff
@@ -746,17 +749,16 @@ The `tracking/revision_history` should be maintained in the backend
...
     Behavior analogous to Semantic Versioning.
-
-When in the `Published` status:
-When switching to `Draft` workflow status, a new Revision History Item is created.
+  - Workflow state changes do not add text to the revision history summary.
 
 When a new document is created on the server, any existing revision history
-items are deleted, a new version number (matching the versioning scheme
-configured on the server) is assigned, and an initial revision history item is created.
+items corresponding to pre-release versions are deleted, a new version number
```
[the remark this answers is not in the document]
Comment (valentindey) [reply documents/architecture-decisions.md]: updated

**Review by cloeser — COMMENTED** [review]
Looking good now.

One thing I noticed: If the workflow state changes to Approved the current release date does not change. Same thing when we go back do Draft.

**Comment by valentindey** [comment]
> Looking good now.
> 
> One thing I noticed: If the workflow state changes to Approved the current release date does not change. Same thing when we go back do Draft.

yes, that part was missing, I updated it

**Review by cloeser — APPROVED** [review]
LGTM now!

**Pull request merged**